File: src/daily_tasks/cli/ticket.py
from __future__ import annotations

from datetime import date

import typer

from daily_tasks.db import session_scope
from daily_tasks.services import tickets as tickets_svc

ticket_app = typer.Typer(help="After a ticket is created, store identifiers on the daily ticket.")


def _parse_date_arg(raw: str) -> date:
    try:
        return date.fromisoformat(raw.strip())
    except ValueError as exc:
        raise typer.BadParameter(f"DATE must be YYYY-MM-DD (got {raw!r}).") from exc


@ticket_app.command("set-ritm")
def ticket_set_ritm(
    ctx: typer.Context,
    entry_date: str = typer.Argument(..., metavar="DATE", help="Ticket day as YYYY-MM-DD."),
    ritm: str = typer.Argument(..., help="RITM / reference number returned by the ticketing system."),
) -> None:
    engine = ctx.obj["engine"]
    d = _parse_date_arg(entry_date)
    with session_scope(engine) as session:
        updated = tickets_svc.set_ritm_for_date(session, entry_date=d, ritm_number=ritm)
    if updated is None:
        typer.secho(f"No ticket found for {d.isoformat()}. Run `log` for that day first.", fg=typer.colors.RED, err=True)
        raise typer.Exit(code=1)
    typer.secho(f"Stored RITM {ritm!r} on {d.isoformat()}.", fg=typer.colors.GREEN)

File: go_version/internal/cli/ticket.go
package cli

import (
	"fmt"

	"daily-tasks/go_version/internal/store"

	"github.com/spf13/cobra"
)

func newTicketCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "ticket",
		Short: "Store identifiers on daily tickets after creation",
	}
	cmd.AddCommand(ticketSetRITMCmd())
	return cmd
}

func ticketSetRITMCmd() *cobra.Command {
	return &cobra.Command{
		Use:   "set-ritm DATE RITM",
		Short: "Store RITM number for a ticket day",
		Args:  cobra.ExactArgs(2),
		RunE: func(cmd *cobra.Command, args []string) error {
			ok, err := store.SetRITMForDate(cmd.Context(), DB, args[0], args[1])
			if err != nil {
				return err
			}
			if !ok {
				return fmt.Errorf("no ticket found for %s; run `log` for that day first", args[0])
			}
			fmt.Printf("Stored RITM %q on %s.\n", args[1], args[0])
			return nil
		},
	}
}
